infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```diff
@@ -84,12 +95,33 @@ private void normalizeMemory(Container container, InternalMachineConfig machineC
     }
   }
 
+  private Map<String, String> toEnv(CheContainer container) {
+    return container.isMountSources()
+        ? Collections.singletonMap(PROJECTS_MOUNT_PATH_ENV_VAR_NAME, PROJECTS_MOUNT_PATH)
+        : null;
+  }
+
   private Map<String, ? extends org.eclipse.che.api.core.model.workspace.config.Volume>
-      toWorkspaceVolumes(List<Volume> volumes) {
+      toWorkspaceVolumes(CheContainer container) throws InfrastructureException {
 
     Map<String, VolumeImpl> result = new HashMap<>();
 
-    for (Volume volume : volumes) {
+    if (container.isMountSources()) {
+      result.put(PROJECTS_VOLUME_NAME, new VolumeImpl().withPath(PROJECTS_MOUNT_PATH));
+    }
+
+    for (Volume volume : container.getVolumes()) {
+      if (volume.getName().equals(PROJECTS_VOLUME_NAME)) {
```

Comment:
According to this current plugins would not allow starting a workspace. Can you elaborate on how are you going to handle that?

Reply:
I thought I would not handle that because since Che7 is not released, we don't give any guarantees on the "API stability".

Reply:
We have Che 7 running on OSIO. So...
And breaking it even temporarily for users that use different versions of Che is also something we might want to avoid

Reply:
@skabashnyuk @ibuziuk we should address that not to break prod

Reply:
This PR, in particular, would not break Che on OSIO. But it would break Che 7 workspaces on nightly Che as far as I understand. And Selenium tests show that Che 7 test doesn't pass.
If we do a change in Theia and rollout that on OSIO without rolling out upstream version with this PR we would break Che on OSIO.